optimum/habana/transformers/training_args.py

```diff
@@ -610,7 +628,7 @@ def _setup_devices(self) -> "torch.device":
 
             if self.use_lazy_mode:
                 logger.info("Enabled lazy mode.")
-            else:
+            elif not self.torch_compile:
                 os.environ["PT_HPU_LAZY_MODE"] = "2"
                 logger.info("Enabled eager mode because use_lazy_mode=False.")
```

Comment:
Please rewrite this part, LAZY_MODE=2 is no longer supported. We should make sure at this point that it's 1 for Lazy (we checked above 0 for compile, so it's no need here).

```
if self.use_lazy_mode:
           assert os.getenv('PT_HPU_LAZY_MODE') == '1' or os.getenv('PT_HPU_LAZY_MODE') == None, f"Argument --use_lazy_mode used, but PT_HPU_LAZY_MODE={os.getenv('PT_HPU_LAZY_MODE')}. For lazy mode, set PT_HPU_LAZY_MODE to 1"
           logger.info("Enabled lazy mode.")
```